app/controllers/settings/ext_cert_controller.rb

```diff
@@ -0,0 +1,260 @@
+    end
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def cert_parse(cert_string)
+    params = {}
+
+    # Check if certificate exists, assume that validation has already occured
+    if !cert_string
+      params[:Notice] = "Certificate not available, please upload a certificate"
+    else
+      cert = read_cert(cert_string)
+      unless cert
+        # :nocov:
+        # Cert has aleady been valiated when entered, an error here would require a failure
+        # in connection from velum to salt or an unintended change in the salt pillar
+        params[:Error] = "Failed to parse stored certificate, please check format and upload again"
+        return params
+        # :nocov:
+      end
+      fingerprint = cert_fingerprint(cert)
+
+      params["Subject Alternative Name".to_sym] = get_san_array(cert)
+      params["Subject Name".to_sym] = cert.subject.to_s.tr("/", " ")
+      params["Issuer Name".to_sym] = cert.issuer.to_s.tr("/", " ")
+      params["Signature Algorithm".to_sym] = cert.signature_algorithm
+      params["Not Valid Before".to_sym] = cert.not_before
+      params["Not Valid After".to_sym] = cert.not_after
+      params[fingerprint[0]] = fingerprint[1]
+    end
+    params
+  end
+
+  def cert_fingerprint(cert)
+    # Error-checking ignored, cert already validated and verified
+    fingerprint = OpenSSL::Digest::SHA256.new(cert.to_der)
+    fingerprint_string = fingerprint.to_s.scan(/../).map(&:upcase).join(":")
+    ["SHA256 Fingerprint".to_sym, fingerprint_string]
+  end
+
+  def key_parse(key_string)
+    params = {}
+    if !key_string
+      params[:Error] = "Key not available, please upload a certificate"
+    # :nocov:
+    # Key has aleady been valiated when entered, an error here would require a failure
+    # in connection from velum to salt or an unintended change in the salt pillar
+    else
+      key = read_key(key_string)
+      unless key
+        params[:Error] = "Failed to parse stored key, please check format and upload again"
+      end
+      key_valid = if key
+        true
+      else
+        false
+      end
+      params["Valid Key"] = key_valid
+      # :nocov:
+    end
+    params
+  end
+
+  # Common method to build cert object from string
+  def read_cert(cert_string)
+    return OpenSSL::X509::Certificate.new(cert_string)
+  rescue OpenSSL::X509::CertificateError, NoMethodError, TypeError
+    # Push error handling to calling method for flexibility
+    return nil
+  end
+
+  # Common method to build key object from string
+  def read_key(key_string)
+    return OpenSSL::PKey::RSA.new(key_string)
+  rescue OpenSSL::PKey::RSAError, NoMethodError, TypeError
+    # Push error handling to calling method for flexibility
+    return nil
+  end
+
+  # Common method to simplify failure renders
+  def render_failure_event(message)
+    set_instance_variables
+    flash[:alert] = message
+    render action: :index, status: :unprocessable_entity
+    false
```

Comment:
Since this always returns false, I'm not sure what the point is here of returning anything.